### PR TITLE
Add ability to update IAM Role Credentials on Configuration

### PR DESCRIPTION
**Pull request by jasonhebert**
<h2>Change Log</h2>
<div id="change-log">
  <h4>Added</h4>
  <ul id="added">
    <li>Added Configuration getter to SellingPartnerSDK.</li>
    <li>Added method to update IAM Role Credentials on Configuration class.</li>
  </ul> 
  <h4>Fixed</h4>  
  <ul id="fixed">
    
  </ul>
  <h4>Changed</h4>
  <ul id="changed">
    
  </ul>  
  <h4>Removed</h4>
  <ul id="removed">
    
  </ul>
  <h4>Deprecated</h4>
  <ul id="deprecated">
    
  </ul>  
  <h4>Security</h4> 
  <ul id="security">
    
  </ul>     
</div>
<hr/>

<h2>Description</h2>

<p>Added the ability to update the IAM Role Credentials on the Configuration for the cases where the session token has expired and new credentials must be provided.</p>

### CODE REVIEW

**Comment by norberttech** [comment]
Don't we need to actually update whole `SellingPartnerSDK` when the session token expires?

**Comment by jasonhebert** [comment]
> Don't we need to actually update whole `SellingPartnerSDK` when the session token expires?

If my logic here is flawed in any way please correct me!

Since we pass a reference to the `Configuration` instance from the `SellingPartnerSDK` instance when instantiating other SDK object instances we can update the properties of the `Configuration` object.

Therefore when the session token expires we can:

1. Get new IAM Role credentials using the `STSClient`
2. Get the `Configuration` instance from the `SellingPartnerSDK` instance
3. Call `updateIAMRoleCredentials` method on the `Configuration` instance to set the new access key, secret key, and session token properties.

This updates the properties on all instances since the SDK instances all have a reference to the `Configuration` instance.

File: src/AmazonPHP/SellingPartner/SellingPartnerSDK.php [hunk 1]
```diff
@@ -84,6 +84,11 @@ public static function create(
         return new self($httpClient, $requestFactory, $streamFactory, $configuration, $logger);
     }
 
+    public function getConfiguration() : Configuration
```
Comment (norberttech) [inline src/AmazonPHP/SellingPartner/SellingPartnerSDK.php]: could you please rename this to `configuration(): Configuration` to follow the convention?

**Comment by norberttech** [comment]
hey @jasonhebert makes sense, thanks for the explanation!

File: src/AmazonPHP/SellingPartner/SellingPartnerSDK.php (hunk 1 above)
Comment (jasonhebert) [reply src/AmazonPHP/SellingPartner/SellingPartnerSDK.php]: Done...

**Pull request merged**